### PR TITLE
Update for latest SDK; Add constructors for DaService and DaVerifier

### PR DESCRIPTION
**Pull request by preston-evans98**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -45,3 +44,6 @@ tendermint-proto = { git = "https://github.com/informalsystems/tendermint-rs.git
 default = ["native"]
 native = ["dep:tokio", "dep:reqwest", "dep:jsonrpsee"]
 verifier = []
+
+[patch.'https://github.com/Sovereign-Labs/sovereign.git']
```
Comment (citizen-stig) [inline Cargo.toml]: I suppose this can be removed, as we merged everything into research branch

Comment (preston-evans98) [reply Cargo.toml]: Good catch! Removed

File: src/types.rs [hunk 2]
```diff
@@ -226,18 +202,16 @@ fn ns_hash_from_b64(input: &str) -> NamespacedHash {
 #[cfg(test)]
 mod tests {
 
-    use nmt_rs::{NamespaceProof, NamespacedSha2Hasher};
-
-    use crate::verifier::ROLLUP_NAMESPACE;
+    // use nmt_rs::{NamespaceProof, NamespacedSha2Hasher};
 
-    use super::{ns_hash_from_b64, RpcNamespacedSharesResponse};
+    // use super::{ns_hash_from_b64, RpcNamespacedSharesResponse};
```
Comment (citizen-stig) [inline src/types.rs:207]: Maybe delete commented code, if we don't need it.

**Review by citizen-stig — APPROVED** [review]

File: src/types.rs (hunk 2 above)
Comment (preston-evans98) [reply src/types.rs:207]: I think Ieft it since we plan to re-enable this test once a Celestia bug is fixed, and we'll need this stuff to enable the test.

**Pull request merged**